vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+    if (status != NC_NOERR) {
+        log_err("Error opening %s", filenames.init_state);
+    }
+
+    // read and validate dimension lengths
+    dimlen = get_nc_dimension(filenames.init_state, "ni");
+    if (dimlen != global_domain.n_nx) {
+        log_err("Number of grid columns in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "nj");
+    if (dimlen != global_domain.n_ny) {
+        log_err("Number of grid rows in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "veg_class");
+    if (dimlen != options.NVEGTYPES) {
+        log_err("Number of veg classes in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "snow_band");
+    if (dimlen != options.SNOW_BAND) {
+        log_err("Number of snow bands in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "nlayer");
+    if (dimlen != options.Nlayer) {
+        log_err("Number of soil layers in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "frost_area");
+    if (dimlen != options.Nfrost) {
+        log_err("Number of frost areas in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "soil_node");
+    if (dimlen != options.Nnode) {
+        log_err("Number of soil nodes in state file does not "
+                "match parameter file");
+    }
+    if (options.LAKES) {
+        dimlen = get_nc_dimension(filenames.init_state, "lake_node");
+        if (dimlen != options.NLAKENODES) {
+            log_err("Number of lake nodes in state file does not "
+                    "match parameter file");
+        }
+    }
+
+    // read dimension variables
+
+    // lat/lon
+    status = nc_inq_varid(nc.nc_id, "lon", &lon_var_id);
+    if (status != NC_NOERR) {
+        log_err("Unable to find variable \"lon\" in %s",
+                filenames.init_state);
+    }
+    status = nc_inq_varid(nc.nc_id, "lat", &lat_var_id);
+    if (status != NC_NOERR) {
+        log_err("Unable to find variable \"lat\" in %s",
+                filenames.init_state);
+    }
+    if (global_domain.info.n_coord_dims == 1) {
+        d1start[0] = 0;
+        dvar = calloc(global_domain.n_nx, sizeof(*dvar));
+        d1count[0] = global_domain.n_nx;
+        status = nc_get_vara_double(nc.nc_id, lon_var_id,
+                                    d1start, d1count, dvar);
+        if (status != NC_NOERR) {
+            log_err("Error reading data from \"lon\" in %s",
+                    filenames.init_state);
+        }
+        for (i = 0; i < global_domain.n_nx; i++) {
+            if (dvar[i] != (double) global_domain.locations[i].longitude) {
```

Comment:
OK, I'll do that.  Should abs_tol and rtol be local vars or constants in a .h file?  I'm a little reluctant to set up constants since they maybe should have context-dependent values (for example, abs_tol of 0.0001 would be too coarse if we get finer than 1/16 degree resolution).  How about for now, just to get the code functioning, I set up local vars within check_state_file() and then a separate issue and PR can deal with how to make them work in a more general way?